packages/flutter/lib/src/widgets/image.dart

```diff
@@ -1006,11 +1006,39 @@ class _ImageState extends State<Image> with WidgetsBindingObserver {
   }
 
   void _resolveImage() {
-    final ImageStream newStream =
-      widget.image.resolve(createLocalImageConfiguration(
+    final DeferringImageProvider<dynamic> provider = DeferringImageProvider<dynamic>(
+      imageProvider: widget.image,
+      getNextAction: () {
+        // Did this element get disposed of (scrolled too far out of view before
+        // scrolling slowed down enough to catch up)?
+        if (!mounted) {
+          return DeferringImageProviderAction.cancel;
+        }
+        // Is it scrolling so fast that the image is unlikely stay around long
+        // enough to be worth decoding? If so, wait until the next task loop
+        // to see if we've slowed down or disappeared.
+        // We're using the large of height or width of the window in physical
+        // pixels. This operation compares physical pixels (the window
+        // dimension) with logical pixels (the scrolling velocity). Using
+        // logical window pixels results in being too aggressive about loading
+        // images, and physical pixels will give us a better approximation of
+        // how much RAM is available.
+        // It's safe to use these calls as point-in-time - do not subscribe to
+        // updates of MediaQuery or Scrollable.
+        final double maxPhysicalPixels = WidgetsBinding.instance.window.physicalSize.longestSide;
+        if ((Scrollable.scrollingVelocityOfContext(context) ?? 0).abs() > maxPhysicalPixels) {
```

Comment:
You are mixing units here, velocity versus physical pixels.  I'd compare velocity to velocity.

Reply:
It's intentional though.  The idea is the DPI is a good heuristic to use for this case.

Reply:
That's fine, i'd document it with another variable like.
`let double fullScreenPerSecond = maxPhysicalPixels;`

Reply:
Is there some scenario where we could end up with enough velocity to meet this condition indefinitely? For example, some sort of absurdly pixel scaling combined with a looping animation?

Reply:
In such a situation we'd have to be scrolling at high velocity indefinitely.  Or, we'd have to be in some kind of crazy scroll activity that's lying to us about how fast it's scrolling indefinitely.

I suppose the most pathological case I can think of is an animation that loops at high velocity such that it scrolls up and down very very fast. If you wanted the image to ever show up in that case, you'd have to precache it.

Reply:
And it couldn't be just any animation, it would have to be an animation that drove a scroll controller such that the activity used was a ballistic activity scrolling at high velocity.